### PR TITLE
ci: pipelines and extensions implementation

### PR DESCRIPTION
**Pull request by joseluis-teran-jalafund**
## Trello Ticket

[Task 02](https://trello.com/c/xIgE9Qgn/9-task-2-ci-workflows-and-vs-code-extensions)

## Description

This PR refers to CI/CD, having two .yml files for our workflow, which has the next jobs; `build` and `test` to automatize the project, and also `check_commits_convention`, `check_branches_convention` for every **push** to **release** branch and `semantic_pull_request` and `check_commit_messages` for each **PR** that will be opened. Finally, the **pull request template** file so that each PR has a description when it's opened. Furthermore of recommending VS Code extensions for project.

## Type of Change

Put an `x` in all the boxes that apply:

- [ ] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [x] 📝 Documentation
- [x] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [ ] My code follows the code style of this project.
- [ ] No existing features have been broken without good reason.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: LosGosusTest/src/Controller/BookControllerTest.cs
Comment (ronaldo-mendoza-jalafund) [inline LosGosusTest/src/Controller/BookControllerTest.cs:1]: Wouldn't it be better to make another PR just for testing instead of integrating it into this PR?

Comment (joseluis-teran-jalafund) [reply LosGosusTest/src/Controller/BookControllerTest.cs:1]: Yeah, you're right. That's why I didn't update the test totally. I just commented to work it later. That's something which I'm currently working on.

Comment (alex-paca-jalafund) [reply LosGosusTest/src/Controller/BookControllerTest.cs:1]: I reviewed and its commented i think for only pass the github filter, so be able to merge, we can merge now the workflow so the test update will be another branch.

**Review by alex-paca-jalafund — APPROVED** [review]
nice.
